QModManager/API/SMLHelper/Utility/EnumCacheManager.cs

```diff
@@ -0,0 +1,210 @@
+﻿namespace QModManager.API.SMLHelper.Utility
+{
+    using QModManager.Utility;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Reflection;
+    using System.Text;
+
+    internal class EnumTypeCache
```

Comment:
Preferably, missing entries should not be deleted.
Missing TechTypes should not be removed from the cache, and SMLHelper should create a dummy item at that index to prevent it from being overridden. Same for CraftTreeTypes

Reply:
I agree that removing missing entries can cause all sorts of problems. Let me propose this:

1. These files are updated with a new format ```{Index}:{Name}:{Active|Inactive}```
2. Whenever a missing custom TechType is detected, we leave the name of the entry alone and instead simply update a bool flag at the end to mark it as inactive.
  a. This flag would be used whenever mods call into the "TryGetModdedTechType" methods,  
to ensure that we don't send back an unusable value.
3. The flag can be switched back to the active state if we see it again on an "AddTechType" call.

This work may have to wait until we've finished merging and be done as another update which may be released along with QMM 3.0 or after it as an update.

Reply:
I agree, the cache system would have to be totally re-written for this.